### PR TITLE
 Update Mockito version to 2.28.2 (fixes #57). 

### LINKED ISSUE #57: Eliminate illegal reflection warning from Mockito on Jdk9+
Post java9, reflective operations like those that at least used to be common in Mockito need an explicit JVM flag set, or else are considered illegal and print as much to stderr. Probably mockito just needs to be upgraded to eliminate this, although I'm probably pretty behind on it.

### PR DESCRIPTION
**Pull request by PascalSchumacher**

### CODE REVIEW

File: src/test/java/testsupport/matchers/ZeroInvocationsMatcher.java [hunk 1]
```diff
@@ -26,7 +27,7 @@ public boolean matches(Object item) {
     @Override
     public void describeMismatch(Object item, final Description description) {
         description.appendText("had these: ");
-        for (Invocation invocation : new MockUtil().getMockHandler(item).getInvocationContainer().getInvocations())
+        for (Invocation invocation : ((InvocationContainerImpl) MockUtil.getMockHandler(item).getInvocationContainer()).getInvocations())
```
Comment (PascalSchumacher) [inline src/test/java/testsupport/matchers/ZeroInvocationsMatcher.java]: Hacky, no idea if there is a better way.

**Comment by jnape** [comment]
Hi @PascalSchumacher! Thanks for the PR!

It looks like there is still a reflective access warning when I run `mvn clean verify` on the `mockito_2` branch:

```log
...
Running com.jnape.palatable.lambda.functions.builtin.fn1.SizeTest
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.mockito.internal.util.reflection.AccessibilityChanger (file:/Users/jnape/.m2/repository/org/mockito/mockito-core/2.28.2/mockito-core-2.28.2.jar) to field java.util.ArrayList.elementData
WARNING: Please consider reporting this to the maintainers of org.mockito.internal.util.reflection.AccessibilityChanger
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
...
```

Are you seeing that on your end as well? I'm currently using oracle-jdk-11.0.2.

File: src/test/java/testsupport/matchers/ZeroInvocationsMatcher.java (hunk 1 above)
Comment (jnape) [reply src/test/java/testsupport/matchers/ZeroInvocationsMatcher.java]: I was able to find `MockUtil#getInvocationHandler`, which does the trick!

**Comment by jnape** [comment]
I fixed the last of the reflection warnings and manually rebased this changeset onto master as of [384a9d5e](https://github.com/palatable/lambda/commit/384a9d5e5473b9a797b9d73faa04b9692ecb0742). Thank you @PascalSchumacher!

**Pull request closed without merging**